superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx

```diff
@@ -16,250 +16,128 @@
-                  {t('See more')}
-                </span>
-              )}
-              {isBodyExpanded && (
-                <>
-                  <br />
-                  {body}
-                  <span
-                    role="button"
-                    tabIndex={0}
-                    className="link"
-                    onClick={() => setIsBodyExpanded(false)}
-                    onKeyDown={event => {
-                      if (event.key === 'Enter') {
-                        setIsBodyExpanded(false);
-                      }
-                    }}
-                  >
-                    {t('See less')}
-                  </span>
-                </>
-              )}
-            </>
-          )}
-        </div>
-      ) : (
-        <ErrorModal
-          level={level}
-          show={isModalOpen}
-          onHide={() => setIsModalOpen(false)}
-          destroyOnClose
-          title={
-            <div className="header">
-              {level === 'error' ? (
-                <Icons.ErrorSolid className="icon" iconColor={iconColor} />
-              ) : (
-                <Icons.WarningSolid className="icon" iconColor={iconColor} />
-              )}
-              <div className="title">{title}</div>
-            </div>
-          }
-          footer={
-            <>
-              {copyText && (
-                <CopyToClipboard
-                  text={copyText}
-                  shouldShowText={false}
-                  wrapped={false}
-                  copyNode={<Button onClick={noOp}>{t('Copy message')}</Button>}
-                />
-              )}
-              <Button
-                cta
-                buttonStyle="primary"
-                onClick={() => setIsModalOpen(false)}
-                tabIndex={0}
-                onKeyDown={event => {
-                  if (event.key === 'Enter') {
-                    setIsModalOpen(false);
-                  }
-                }}
-              >
-                {t('Close')}
-              </Button>
-            </>
-          }
-        >
-          <>
-            <p>{subtitle}</p>
-            {/* This break was in the original design of the modal but
-            the spacing looks really off if there is only
-            subtitle or a body */}
-            {subtitle && body && <br />}
-            {body}
-          </>
-        </ErrorModal>
+    </div>
+  );
+
+  const renderAlert = () => (
+    <Alert
+      description={renderDescription()}
+      type={type}
+      showIcon
+      closable={closable}
+    >
+      <strong>{errorType}</strong>
+      {message && (
+        <>
+          : <span>{message}</span>
+        </>
       )}
-    </ErrorAlertDiv>
+    </Alert>
   );
-}
+
+  if (compact) {
+    return (
+      <>
+        <Tooltip title={`${errorType}: ${message}`}>
+          <span role="button" onClick={() => setShowModal(true)} tabIndex={0}>
+            {renderTrigger()}
+          </span>
+        </Tooltip>
+        <Modal
```

Comment:
Should we remove the ability to close the alert within a Modal? It seems odd to be able to close an alert which is within a Modal container.

<img width="760" alt="Screenshot 2025-01-15 at 17 39 58" src="https://github.com/user-attachments/assets/faeba9a9-c931-4a69-8a56-07063ad0d577" />

Reply:
Probably, would be nice to force this in this context, though the parent component can just pass the prop too. That may be a thing I didn't fully sort out here -> making sure close is only available where it make sense. Might be worth a sweep reviewing everywhere `ErrorAlert` is use and double checking that we set this right in that context.

Reply:
Ok i added a line of logic to handle this